### PR TITLE
Allow diff generation to consider whitespace

### LINKED ISSUE #34: InlineDiffBuilder needs an overload that takes ignoreCase parameter (and probably ignoreWhitespaces, too)
InlineDiffBuilder.BuildDiffModel calls into IDiffer.CreateLineDiffs(oldText, newText, true), therefore hardcoding ignoreCase parameter to a default value, which is false, and also hardcoding ignoreWhitespace to true (per call done above). 
An overload to BuildDiffModel is needed to pass in ignoreCase (and probably ignoreWhitespace) parameters.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #34

### CODE REVIEW

File: DiffPlex/DiffBuilder/Model/DiffPiece.cs [hunk 1]
```diff
@@ -29,5 +30,48 @@ public DiffPiece()
...
+                && SubPiecesEqual(other);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = 1688038063;
+            hashCode = hashCode * -1521134295 + Type.GetHashCode();
+            hashCode = hashCode * -1521134295 + EqualityComparer<int?>.Default.GetHashCode(Position);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Text);
+            hashCode = hashCode * -1521134295 + EqualityComparer<int?>.Default.GetHashCode(SubPieces?.Count);
```
Comment (sharwell) [inline DiffPlex/DiffBuilder/Model/DiffPiece.cs]: 📝 Deep hash code is not required for correctness.

File: Facts.DiffPlex/SideBySideDiffBuilderFacts.cs [hunk 2]
```diff
@@ -367,6 +367,170 @@ public void Will_build_diffModel_for_partially_different_lines()
...
+                Assert.Equal(
+                    model.OldText.Lines,
+                    new DiffPiece[]
+                    {
+                        new DiffPiece("1", ChangeType.Unchanged, 1),
+                        new DiffPiece(" 2", ChangeType.Modified, 2)
+                        {
+                            SubPieces =
+                            {
+                                new DiffPiece("", ChangeType.Deleted, 1),
```
Comment (sharwell) [inline Facts.DiffPlex/SideBySideDiffBuilderFacts.cs]: :memo: This piece seemed strange to me, but this behavior was not introduced by my change so I didn't do work to fix it at the same time.

**Comment by sharwell** [comment]
@mmanela I don't have access to the build results to see what the problem(s) are.

**Comment by mmanela** [comment]
@sharwell Thanks for pointing that out, I just switched the build to another project that is publicly accessible. Triggering it to pick up this pr and lets see 
/AzurePipelines run

**Comment by sharwell** [comment]
@mmanela thanks. This should be an easy fix in the morning

**Comment by sharwell** [comment]
@mmanela This is now fixed

**Pull request merged**